sys/hashes/sha256.c

```diff
@@ -318,3 +318,71 @@ const unsigned char *hmac_sha256(const unsigned char *key,
 
     return result;
 }
+
+unsigned char *sha256_chain(const unsigned char *seed, size_t seed_length,
+                            size_t elements,
+                            unsigned char *tail_element)
+{
+    unsigned char tmp1[SHA256_DIGEST_LENGTH];
+    unsigned char tmp2[SHA256_DIGEST_LENGTH];
+
+    /* if no sha256-chain can be created return NULL */
+    if (elements < 2) {
+        return NULL;
+    }
+
+    /* 1st iteration */
+    sha256(seed, seed_length, tmp2);
+    
+    for (size_t i = 1; i < elements-1; ++i) {
+        if ( (i % 2) == 0 ) {
+            sha256(tmp1, SHA256_DIGEST_LENGTH, tmp2);
+        }
+        else {
+            sha256(tmp2, SHA256_DIGEST_LENGTH, tmp1);
+        }
+    }
+
+    /* last iteration */
+    if ( (elements % 2) == 0 ) {
+        sha256(tmp2, SHA256_DIGEST_LENGTH, tail_element);
+    }
+    else {
+        sha256(tmp1, SHA256_DIGEST_LENGTH, tail_element);
+    }
+    return tail_element;
+}
+
+int verify_sha256_chain_element(unsigned char *element,
+                                 size_t element_index,
+                                 unsigned char *tail_element,
+                                 size_t chain_length)
+{
+    unsigned char tmp1[SHA256_DIGEST_LENGTH];
+    unsigned char tmp2[SHA256_DIGEST_LENGTH];
+
+    int delta_count = (chain_length - element_index);
+
+    /* if we have an index mismatch return false */
+    if ( delta_count < 1) {
+        return -1;
```

Comment:
I guess I do something wrong the `assert`s doesn't seem to abort executing the unittests,
No matter if I set them right, i.e. `assert(delta_count >= 1);` or wrong, i.e. `assert(delta_count < 1);`